### ISSUE TITLE
[Accessibility]: Clunky code intelligence policy repository pattern deletion

### ISSUE THREAD

**Issue by efritz** [sourcegraph/sourcegraph#35194]
### Audit type

Screen reader navigation

### User journey audit issue

#33518

### Problem description

<img width="884" alt="typeahead" src="https://user-images.githubusercontent.com/103087/167508039-88541acf-9bd3-4281-b3fc-3fbf214a94c5.png">

Deleting a repository pattern (when editing the site-admin code intelligence policy) does not put the user's focus on an control after the button is pressed. Additionally this button does not have an aria description that makes sense. I'm unaware it's a delete button apart from having put it there myself.

### Expected behavior

The screen reader announces a change in the rendered page.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by umpox** [comment]
@efritz I think we now can close this, LMK if you disagree!

> Deleting a repository pattern (when editing the site-admin code intelligence policy) does not put the user's focus on an control after the button is pressed.

From my testing, the focus isn't lost (pressing tab then navigates to the next field available in the form). I think this is generally OK, AFAIK automatically navigating to a new control wouldn't be required here unless it was particularly relevant to the immediate action that was just performed.

> Additionally this button does not have an aria description that makes sense. I'm unaware it's a delete button apart from having put it there myself.

This should now be fixed, the button has an `aria-label="Delete the repository pattern"`
